cpp/src/gandiva/gdv_string_function_stubs.cc

```diff
@@ -264,6 +264,356 @@ const char* gdv_fn_lower_utf8(int64_t context, const char* data, int32_t data_le
   return out;
 }
 
+//converts utf8 to utf16 encoding
+GANDIVA_EXPORT
+const char* gdv_fn_encode(int64_t context, const char* data, int32_t data_len,
+                          const char* charset, int32_t charset_len, int32_t* out_len) { 
+  if (data_len == 0) {
+    *out_len = 0;
+    return "";
+  }
+  
+  char* out = reinterpret_cast<char*>(
+    gdv_fn_context_arena_malloc(context, 32 * data_len));
+  if (out == nullptr) {
+    gdv_fn_context_set_error_msg(context,
+    "Could not allocate memory for output string");
+    *out_len = 0;
+    return "";
+  }
+
+  int32_t char_len;
+  uint32_t char_codepoint;
+  
```

Comment:
The code seems to have some format problems(it can give errors on some CI steps). To fix it, install clang-format v12 in your machine and run that command:
```
python3 [PATH-TO_ARROW]/arrow/cpp/build-support/run_clang_format.py --clang_format_binary /usr/bin/clang-format-12 --exclude_globs [PATH_TO_ARROW]/arrow/cpp/build-support/lint_exclusions.txt --source_dir [PATH-TO_ARROW]/arrow/cpp/src --fix
```

Change the PATH_TO_ARROW to the location of your arrow repository inside your machine. And the `/usr/bin/clang-format-12` is the path were clang-format is installed in Linux machines, maybe for Mac and Windows the path is different

Reply:
You can test to check if your code has format problems locally following that command: https://github.com/apache/arrow/blob/3d6240c1ee7802829d2ed209f4135906e9413915/docker-compose.yml#L1650